pkg/envoy/sds/builder.go

```diff
@@ -0,0 +1,167 @@
+package sds
+
+import (
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_auth "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/secrets"
+	"github.com/openservicemesh/osm/pkg/errcode"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+type sdsBuilder struct {
+	proxy *envoy.Proxy
+
+	// Service certificate for this proxy
+	serviceCert *certificate.Certificate
+
+	// The DiscoveryRequest contains the certs to generate secrets for
+	// requestedCerts is expected to be a list of either "service-cert:namespace/service" or "root-cert:namespace/service"
+	requestedCerts []string
+
+	trustDomain string
+
+	serviceIdentitiesForOutboundServices map[service.MeshService][]identity.ServiceIdentity
+}
+
+func NewBuilder() *sdsBuilder {
```

Comment:
The linter expects exported functions and types to have a comment, so you need to add those for the linter to pass.